pkg/kubeadm/master/manifests.go

```diff
@@ -0,0 +1,194 @@
+
+func WriteStaticPodManifests(params *kubeadmapi.BootstrapParams) error {
+	staticPodSpecs := map[string]api.Pod{
+		// TODO this needs a volume
+		"etcd": componentPod(api.Container{
+			Command: []string{
+				"/usr/local/bin/etcd",
+				"--listen-peer-urls=http://127.0.0.1:2380",
+				"--addr=127.0.0.1:2379",
+				"--bind-addr=127.0.0.1:2379",
+				"--data-dir=/var/etcd/data",
+			},
+			Image:         "gcr.io/google_containers/etcd:2.2.1", // TODO parametrise
+			LivenessProbe: componentProbe(2379, "/health"),
+			Name:          "etcd-server",
+			Resources:     componentResources("200m"),
+		}),
+		// TODO bind-mount certs in
+		"kube-apiserver": componentPod(api.Container{
+			Name:  "kube-apiserver",
+			Image: params.EnvParams["hyperkube_image"],
+			Command: []string{
+				"/hyperkube",
+				"apiserver",
+				"--address=127.0.0.1",
+				"--etcd-servers=http://127.0.0.1:2379",
+				"--cloud-provider=fake", // TODO parametrise
+				"--admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,ResourceQuota",
+				SERVICE_CLUSTER_IP_RANGE,
+				"--service-account-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--client-ca-file=/etc/kubernetes/pki/ca.pem",
+				"--tls-cert-file=/etc/kubernetes/pki/apiserver.pem",
+				"--tls-private-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--secure-port=443",
+				"--allow-privileged",
+				COMPONENT_LOGLEVEL,
+				"--token-auth-file=/etc/kubernetes/pki/tokens.csv",
+			},
+			VolumeMounts:  []api.VolumeMount{pkiVolumeMount()},
+			LivenessProbe: componentProbe(8080, "/healthz"),
+			Resources:     componentResources("250m"),
+		}, pkiVolume(params)),
+		"kube-controller-manager": componentPod(api.Container{
+			Name:  "kube-controller-manager",
+			Image: params.EnvParams["hyperkube_image"],
+			Command: []string{
+				"/hyperkube",
+				"controller-manager",
+				MASTER,
+				CLUSTER_NAME,
+				"--root-ca-file=/etc/kubernetes/pki/ca.pem",
+				"--service-account-private-key-file=/etc/kubernetes/pki/apiserver-key.pem",
+				"--cluster-signing-cert-file=/etc/kubernetes/pki/ca.pem",
+				"--cluster-signing-key-file=/etc/kubernetes/pki/ca-key.pem",
+				"--insecure-experimental-approve-all-kubelet-csrs-for-group=system:kubelet-bootstrap",
+				COMPONENT_LOGLEVEL,
+			},
+			VolumeMounts:  []api.VolumeMount{pkiVolumeMount()},
+			LivenessProbe: componentProbe(10252, "/healthz"),
+			Resources:     componentResources("200m"),
+		}, pkiVolume(params)),
+		"kube-scheduler": componentPod(api.Container{
+			Name:  "kube-scheduler",
+			Image: params.EnvParams["hyperkube_image"],
```

Comment:
Ok, this sounds simple enough.

On Fri, 2 Sep 2016, 20:57 Lucas Käldström, <notifications@github.com> wrote:

> In pkg/kubeadm/master/manifests.go
> <https://github.com/kubernetes/kubernetes/pull/31221#discussion_r77403637>
> :
>
> > +				MASTER,
> > +				CLUSTER_NAME,
> > +				"--root-ca-file=/etc/kubernetes/pki/ca.pem",
> > +				"--service-account-private-key-file=/etc/kubernetes/pki/apiserver-key.pem",
> > +				"--cluster-signing-cert-file=/etc/kubernetes/pki/ca.pem",
> > +				"--cluster-signing-key-file=/etc/kubernetes/pki/ca-key.pem",
> > +				"--insecure-experimental-approve-all-kubelet-csrs-for-group=system:kubelet-
>
> bootstrap",
> > +				COMPONENT_LOGLEVEL,
> > +			},
> > +			VolumeMounts:  []api.VolumeMount{pkiVolumeMount()},
> > +			LivenessProbe: componentProbe(10252, "/healthz"),
> > +			Resources:     componentResources("200m"),
> > +		}, pkiVolume(params)),
> > +		"kube-scheduler": componentPod(api.Container{
> > +			Name:  "kube-scheduler",
> > +			Image: params.EnvParams["hyperkube_image"],
>
> I think a switch of some kind would be just fine e.g. (pseudocode)
> if KUBE_INDIVIDUAL_IMAGES=true; then
> env["scheduler-image"] = "gcr.io/.../kube-scheduler...
> env["apiserver-image"] = "gcr.io/.../kube-apiserver...
> else
> env["scheduler-image"] = "gcr.io/.../hyperkube...
> env["apiserver-image"] = "gcr.io/.../hyperkube...
> fi
>
> That would matter a lot to me
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/31221/files/072ad063150a933a98fe8317707ebeac49d1239f#r77403637>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPWS_M6QqXbEjFsF5xP20X9xO4Z5sndks5qmH-TgaJpZM4JqsTH>
> .
>